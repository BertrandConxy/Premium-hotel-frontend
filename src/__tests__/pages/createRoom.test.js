import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import CreateRoom from '../../pages/CreateRoom';
import store from '../../redux/configureStore';

const CreateRoomProvider = () => (
  <Provider store={store}>
    <Router>
      <CreateRoom />
    </Router>
  </Provider>
);

describe('render component', () => {
  it('renders the rooms list form', () => {
    render(<CreateRoomProvider />);
    const roomName = screen.getByLabelText(/room name/i);
    const city = screen.getByLabelText(/city/i);
    const roomType = screen.getByLabelText(/Room type/i);
    const submit = screen.getByText(/submit/i);
    expect(roomName).toBeInTheDocument();
    expect(city).toBeInTheDocument();
    expect(roomType).toBeInTheDocument();
    expect(submit).toBeInTheDocument();
  });
});

describe('Snapshot test', () => {
  it('looks the same as the snapshot', () => {
    const component = renderer.create(<CreateRoomProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
