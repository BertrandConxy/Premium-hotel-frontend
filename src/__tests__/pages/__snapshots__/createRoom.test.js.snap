// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<section
  className="form-container"
>
  <button
    className="p-2 vis btn"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars text-white"
      data-icon="bars"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="nav pt-10"
  >
    <img
      alt="Premium Hotel Logo"
      className="lunar-logo"
      src="lunar.png"
    />
    <section>
      <nav>
        <ul>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              aria-current="page"
              className="active"
              exact="true"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/reservations"
              onClick={[Function]}
            >
              My Reservations
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/create_room"
              onClick={[Function]}
            >
              Create Room
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/rooms"
              onClick={[Function]}
            >
              Delete Room
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/add_reservation"
              onClick={[Function]}
            >
              Reserve Room
            </a>
          </li>
        </ul>
      </nav>
    </section>
  </div>
  <div
    className="my_container container"
  >
    <form
      className="mx-auto row row-cols-1 row-cols-lg-2 bg-white p-4 d-flex justify-content-center align-items-center form"
      onSubmit={[Function]}
    >
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicName"
        >
          Room Name
        </label>
        <input
          className="form-control"
          id="formBasicName"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
      </div>
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicCity"
        >
          City
        </label>
        <select
          aria-label="Select Hotel City Field"
          className="form-select"
          id="formBasicCity"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option>
            Select Hotel City
          </option>
          <option
            value="Lusaka"
          >
            Lusaka
          </option>
          <option
            value="Abuja"
          >
            Abuja
          </option>
          <option
            value="Lagos"
          >
            Lagos
          </option>
          <option
            value="Morocco"
          >
            Morocco
          </option>
        </select>
      </div>
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicRate"
        >
          Rate($)
        </label>
        <input
          className="form-control"
          id="formBasicRate"
          name="rate"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter rate"
          type="number"
          value={0}
        />
      </div>
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicRoomType"
        >
          Room Type
        </label>
        <select
          aria-label="Select Room Type Field"
          className="form-select"
          id="formBasicRoomType"
          name="room_type"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option>
            Select Room Type
          </option>
          <option
            value="Single"
          >
            Single
          </option>
          <option
            value="Double"
          >
            Double
          </option>
          <option
            value="Mini Suite"
          >
            Mini Suite
          </option>
          <option
            value="Master Suite"
          >
            Master Suite
          </option>
          <option
            value="Personalized plant setups"
          >
            Executive Suite
          </option>
          <option
            value="Presidential Suite"
          >
            Presidential Suite
          </option>
        </select>
      </div>
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicAmenities"
        >
          Choose An Amenity
        </label>
        <select
          aria-label="Select Amenity Field"
          className="form-select"
          id="formBasicAmenities"
          name="amenities"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option>
            Select Your Unique Amenity
          </option>
          <option
            value="In-Room Cocktail Station"
          >
            In-Room Cocktail Station
          </option>
          <option
            value="In-room games"
          >
            In-room games
          </option>
          <option
            value="In-room workout equipment"
          >
            In-room workout equipment
          </option>
          <option
            value="Outdoor dining"
          >
            Outdoor dining
          </option>
          <option
            value="Executive Suite"
          >
            Personalized plant setups
          </option>
          <option
            value="Customized Room Theme"
          >
            Customized Room Theme
          </option>
        </select>
      </div>
      <div
        className="col mb-3"
      >
        <label
          className="form-label"
          htmlFor="formBasicFile"
        >
          Add Your Image Link
        </label>
        <input
          className="form-control"
          id="formBasicFile"
          name="picture"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Add an image URL"
          type="text"
          value=""
        />
      </div>
      <div
        className="w-md-75 mx-auto d-flex justify-content-center align-items-center mt-4"
      >
        <button
          className="w-50 mx-auto room-btn btn btn-primary"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</section>
`;
